### PR TITLE
Add AutoParse baseline and module 'langchain_parse' and 'clova'

### PR DESCRIPTION
**Pull request by bwook00**
Plan to write docs after code review ^^

### CODE REVIEW

File: tests/autorag/data/parse/test_clova_ocr.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+
+def test_extract_text_from_fields():
+	result_path = os.path.join(data_dir, "clova_data", "result_sample.json")
+	with open(result_path, "r", encoding="utf-8") as file:
+		result = json.load(file)
+
+	text_path = os.path.join(data_dir, "clova_data", "result_text.txt")
+	with open(text_path, "r", encoding="utf-8") as file:
+		expect_text = file.read()
```
Comment (vkehfdl1) [inline tests/autorag/data/parse/test_clova_ocr.py:85]: You can make "result_sample.json" and "result_text.txt" to the pytest fixture

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]

File: autorag/parser.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	) -> Tuple[List[Callable], List[Dict]]:
+		module_callable_list, module_params_list = [], []
+		for module in modules:
+			module_instance = Module.from_dict(module)
+			module_params_list.append(module_instance.module_param)
+			module_callable_list.append(module_instance.module)
+
+		combinations = list(map(make_combinations, module_params_list))
+		module_list, combination_list = explode(module_callable_list, combinations)
+		return module_list, combination_list
```
[the remark this answers is not in the document]
Comment (bwook00) [reply autorag/parser.py:94]: This function comes from `Node` class `get_param_combinatios`, but it is little bit different code! So I think this function should be here! how do you think about this?

[the remark this answers is not in the document]
Comment (vkehfdl1) [reply autorag/parser.py:94]: I didn't notice the difference

File: autorag/data/parse/clova.py [hunk 3]
```diff
@@ -90,22 +89,31 @@ async def clova_ocr_pure(
...
+		"""Generate image names based on the PDF file name and the number of pages."""
+		pdf_name = _pdf_path.split("/")[-1]
+		pure_pdf_name = pdf_name.split(".pdf")[0]
+		image_name_lst = [
+			f"{pure_pdf_name}_{page_num + 1}.png" for page_num in range(num_pages)
+		]
+		return image_name_lst
+
+	image_data_list = convert_pdf_to_images(pdf_path)
+	image_name_list = generate_image_names(pdf_path, len(image_data_list))
```
Comment (vkehfdl1) [inline autorag/data/parse/clova.py:116]: Just delete pdf_to_images and use below
```
image_data_list = convert_pdf_to_images(pdf_path)
image_name_list = generate_image_names(pdf_path, len(image_data_list))
```
at clove_pure function?

**Review by vkehfdl1 — CHANGES_REQUESTED** [review]

**Review by vkehfdl1 — APPROVED** [review]
LGTM

**Pull request merged**